backend/layers/persistence/persistence.py

```diff
@@ -840,6 +842,16 @@ def get_dataset_artifacts_by_version_id(self, dataset_version_id: DatasetVersion
             )
         return self.get_dataset_artifacts(artifact_ids[0])
 
+    def get_artifact_by_uri_suffix(self, uri_suffix: str) -> Optional[DatasetArtifact]:
+        """
+        Returns the artifact with the given uri suffix
+        """
+        with self._manage_session() as session:
+            artifact = session.query(DatasetArtifactTable).filter(DatasetArtifactTable.uri.endswith(uri_suffix)).first()
+            if artifact:
+                return self._row_to_dataset_artifact(artifact)
+            return None
```

Comment:
This seems like a very permissive approach to me. What is the use-case where there is more than one entry that matches the query, but it's fine to return whichever?